media/mod_sampledata/js/sampledata-process.js

```diff
@@ -3,78 +3,99 @@
-				plugin: 'SampledataApplyStep' + step,
-				step: step
-			}
-		});
-		request.done(function(response){
-			$stepLi.children('.loader-image').remove();
-
-			if (response.success && response.data && response.data.length > 0) {
-				var success, value, resultClass, $msg;
-
-				// Display all messages that we got
-				for(var i = 0, l = response.data.length; i < l; i++) {
-					value   = response.data[i];
-					success = value.success;
-					resultClass = success ? 'success' : 'error';
-					$stepLi.append($('<div>', {
-						html: value.message,
-						'class': 'alert alert-' + resultClass,
-					}));
-				}
+!(function (Joomla, document) {
+	'use strict';
 
-				// Update progress
-				$progress.val(step/steps);
+	var inProgress = false;
 
-				// Move on next step
-				if (success) {
-					step++;
-					sampledataAjax(type, steps, step);
+	Joomla.sampledataAjax = function(type, steps, step) {
+		// Get options
+		var options = Joomla.getOptions('sample-data');
+
+		// Create list
+		var list = document.createElement('li');
+		list.classList.add('sampledata-steps-' + type + '-' + step);
+
+		// Create paragraph
+		var para = document.createElement('p');
+		para.classList.add('loader-image');
+		para.classList.add('text-center');
+
+		// Create image
+		var img = document.createElement('img');
+		img.setAttribute('src', options.icon);
+		img.setAttribute('width', 30);
+		img.setAttribute('height', 30);
+
+		// Append everything
+		para.appendChild(img);
+		list.appendChild(para);
+		document.querySelector('.sampledata-progress-' + type + ' ul').appendChild(list);
+
+		Joomla.request({
+			url: options.url + '&type=' + type + '&plugin=SampledataApplyStep' + step + '&step=' + step,
+			method: 'GET',
+			perform: true,
+			onSuccess: function(response, xhr) {
+				var response = JSON.parse(response);
+				// Remove loader image
+				var loader = list.querySelector('.loader-image');
+				loader.parentNode.removeChild(loader);
+
+				if (response.success && response.data && response.data.length > 0) {
+					var success, value, progressClass;
+					var progress = document.querySelector('.sampledata-progress-' + type + ' .progress-bar');
+
+					// Display all messages that we got
+					for (var i = 0, l = response.data.length; i < l; i++) {
+						value   = response.data[i];
+						success = value.success;
+						progressClass = success ? 'bg-success' : 'bg-danger';
+
+						// Display success alert
+						if (success) {
+							Joomla.renderMessages({success: [value.message]}, '.sampledata-steps-' + type + '-' + step);
+						} else {
+							Joomla.renderMessages({error: [value.message]}, '.sampledata-steps-' + type + '-' + step);
+						}
+					}
+
+					// Update progress
+					progress.innerText = step + '/' + steps;
+					progress.style.width = (step/steps) * 100 + '%';
+					progress.classList.add(progressClass);
+
+					// Move on next step
+					if (success) {
+						step++;
+						Joomla.sampledataAjax(type, steps, step);
```

Comment:
This will fire requests as long as none of them fails (and it will render an error message)... 
We need a check like (currentStep <= totalSteps)